sql/driver/driver_test.go

```diff
@@ -383,15 +449,29 @@ func TestInsertSelectDelete(t *testing.T) {
 			t.Fatal(err)
 		} else {
 			results := readAll(t, rows)
-			expectedResults := [][]string{
+			expected := [][]string{
 				{"k", "v"},
 				{"a", "b"},
 				{"c", "d"},
 				{"e", "f"},
 				{"f", "g"},
 			}
-			if !reflect.DeepEqual(expectedResults, results) {
-				t.Fatalf("expected %s, but got %s", expectedResults, results)
+			expectedResultSlice := asResultSlice(expected)
+
+			nilStr := "nil"
+			rowWithNil := []*string{&nilStr, nil}
+
+			// Because of the primary keys used here, the natural ordering of
+			// the results varies in the two schemas.
```

Comment:
what exactly makes the difference? Specifying the PK after the constraint or that the PK is `(k,v)` instead of just `k`?